<launch>
  <arg name="cameras" default="true"/>

  <rosparam command="load" file="$(find mrover)/config/esw.yaml"/>

  <node pkg="nodelet" type="nodelet" name="perception_nodelet_manager" required="true"
        args="manager" output="screen"/>

  <include file="$(find mrover)/launch/base.launch"/>

  <include file="$(find mrover)/launch/jetson_can.launch"/>

  <group if="$(arg cameras)">
    <include file="$(find mrover)/launch/jetson_base_cameras.launch"/>
  </group>

  <node name="drive_left_bridge" pkg="mrover" type="simple_bridge" output="screen">
    <param name="group_name" value="drive_left"/>
  </node>
  <node name="drive_right_bridge" pkg="mrover" type="simple_bridge" output="screen">
    <param name="group_name" value="drive_right"/>
  </node>

  <node name="imu_driver" pkg="mrover" type="imu_driver.py"
        respawn="true" respawn_delay="2"
        output="screen"/>

  <node name="gps_driver" pkg="mrover" type="rover_gps_driver.py"
        respawn="true" respawn_delay="2"
        output="screen"/>

  <node name="arduino_led_hw_bridge" pkg="mrover" type="arduino_led_hw_bridge.py" output="screen"/>

  <node name="mast_gimbal_bridge" pkg="mrover" type="simple_bridge" output="screen">
    <param name="group_name" value="mast_gimbal"/>
  </node>
</launch>
